File: front/src/types/subscription/subscription.ts
export const SubscriptionPlan = {
    Starter: 'Starter',
    Pro: 'Pro',
    Ultra: 'Ultra',
} as const;

export type SubscriptionPlan = (typeof SubscriptionPlan)[keyof typeof SubscriptionPlan];

export const SubscriptionStatus = {
    Active: 'Active',
    Cancelled: 'Cancelled',
    Expired: 'Expired',
} as const;

export type SubscriptionStatus = (typeof SubscriptionStatus)[keyof typeof SubscriptionStatus];

export interface Subscription {
    id: string;
    userId: string;
    plan: SubscriptionPlan;
    status: SubscriptionStatus;
    startDate: string;
    endDate: string | null;
    cancelledAt: string | null;
}

File: backend/VideoBackend.Domain/Entities/Subscription/Subscription.cs
using VideoBackend.Domain.Enums;

namespace VideoBackend.Domain.Entities.Subscription;

public class Subscription
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public SubscriptionPlan Plan { get; set; }
    public SubscriptionStatus Status { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime? CancelledAt { get; set; }
    public string? ExternalId { get; set; }
}
